pkg/orchestrator/selection/discovery/chained.go

```diff
@@ -1,63 +0,0 @@
-package discovery
-
-import (
-	"context"
-	"errors"
-
-	"github.com/bacalhau-project/bacalhau/pkg/models"
-	"github.com/bacalhau-project/bacalhau/pkg/orchestrator"
-	pkgerrors "github.com/pkg/errors"
-	"github.com/rs/zerolog/log"
-	"golang.org/x/exp/maps"
-)
-
-type Chain struct {
-	discoverers  []orchestrator.NodeDiscoverer
-	ignoreErrors bool
-}
```

Comment:
I remember you saying at one point that you are removing all chain implementations and replacing them with simple arrays. While the Chain looks like a simple array, it implements `NodeDiscoverer` interface which allows different components to just accept the interface rather than an array of interfaces. This simplifies testing, error handling (e.g. `ignoreErrors`), and can allow chain implementations with `AND` or `OR` logic.